streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -115,6 +182,36 @@ class ZStream[-R, +E, -M, +B, +A](
       _ <- pull.toManaged_
     } yield ()
 
+  /**
+   * Runs the sink on the stream to produce either the sink's internal state or an error.
+   */
+  def runQuery[R1 <: R, E1 >: E, A1 >: A, B](sink: ZSink[R1, E1, B, A1, Nothing]): ZIO[R1, E1, B] =
```

Comment:
I think we can allow any sink here, not just sinks that don't terminate. I'd say we need to change the `Nothing` to `Any`.

Reply:
Yeah I had that initially, just wondered whether this is something we'd want to impose, but you're right it is clear from the type at we never use the output anyway.